webauthn4j-core/src/main/java/com/webauthn4j/server/ServerProperty.java

```diff
@@ -30,31 +34,47 @@
     // ~ Instance fields
     // ================================================================================================
 
-    private final Origin origin;
+    private final Set<Origin> origins;
     private final byte[] tokenBindingId;
 
     // ~ Constructor
     // ========================================================================================================
 
     public ServerProperty(Origin origin, String rpId, Challenge challenge, byte[] tokenBindingId) {
+        this(origin != null ? Collections.singleton(origin) : Collections.emptySet(),
+                rpId,
+                challenge, tokenBindingId);
+    }
+
+    public ServerProperty(Collection<Origin> origins, String rpId, Challenge challenge, byte[] tokenBindingId) {
         super(rpId, challenge);
-        this.origin = origin;
+        this.origins = (origins != null && !origins.isEmpty()) ? Collections.unmodifiableSet(new HashSet<>(origins)) : Collections.emptySet();
         this.tokenBindingId = tokenBindingId;
     }
 
     // ~ Methods
     // ========================================================================================================
 
     /**
-     * Returns the {@link Origin}
+     * Returns a single {@link Origin}, provided that this ServerProperty is configured with only a single origin
      *
      * @return the {@link Origin}
      */
     public Origin getOrigin() {
-        return origin;
+        final int originsSize = origins.size();
+        switch (originsSize){
+            case 0:
+                return null;
+            case 1:
+                return origins.iterator().next();
+            default:
+                throw new IllegalStateException("There are multiple Origins associated with this ServerProperty");
+        }
     }
```

Comment:
This method is better to be removed in the future release. I added deprecated annotation in the follow-up commit. https://github.com/webauthn4j/webauthn4j/pull/354/commits/ac64084dc28d1b57084043f65548161cfa849f4a